server/poppler-hack.cc

```diff
@@ -33,90 +30,4 @@ GType poppler_annot_markup_get_type (void) G_GNUC_CONST;
   (G_TYPE_CHECK_INSTANCE_TYPE ((obj), POPPLER_TYPE_ANNOT_MARKUP))
 #define POPPLER_TYPE_ANNOT_MARKUP (poppler_annot_markup_get_type ())
 
-#if POPPLER_CHECK_VERSION(0,72,0)
-#define GET_CSTR c_str
-#else
-#define GET_CSTR getCString
-#endif
-
-  struct PopplerAnnot
-  {
-    GObject  parent_instance;
-    Annot   *annot;
-  };
-
-  struct PopplerAnnotMarkup
-  {
-    GObject  parent_instance;
-  };
-
-  struct PopplerRectangle
-  {
-    double x1;
-    double y1;
-    double x2;
-    double y2;
-  };
-
-  // This function does not modify its argument s, but for
-  // compatibility reasons (e.g. getLength in GooString.h before 2015)
-  // with older poppler code, it can't be declared as such.
-  char *_xpoppler_goo_string_to_utf8(/* const */ GooString *s)
-  {
-    char *result;
-
-    if (! s)
-      return NULL;
-
-    if (s->hasUnicodeMarker()) {
-      result = g_convert (s->GET_CSTR () + 2,
-                          s->getLength () - 2,
-                          "UTF-8", "UTF-16BE", NULL, NULL, NULL);
-    } else {
-      int len;
-      gunichar *ucs4_temp;
-      int i;
-
-      len = s->getLength ();
-      ucs4_temp = g_new (gunichar, len + 1);
-      for (i = 0; i < len; ++i) {
-        ucs4_temp[i] = pdfDocEncoding[(unsigned char)s->getChar(i)];
-      }
-      ucs4_temp[i] = 0;
-
-      result = g_ucs4_to_utf8 (ucs4_temp, -1, NULL, NULL, NULL);
-
-      g_free (ucs4_temp);
-    }
-
-    return result;
-  }
-#ifdef HAVE_POPPLER_ANNOT_WRITE
-  // Set the rectangle of an annotation.  It was first added in v0.26.
-  void xpoppler_annot_set_rectangle (PopplerAnnot *a, PopplerRectangle *rectangle)
-  {
-    GooString *state = (GooString*) a->annot->getAppearState ();
-    char *ustate = _xpoppler_goo_string_to_utf8 (state);
-
-    a->annot->setRect (rectangle->x1, rectangle->y1,
-                       rectangle->x2, rectangle->y2);
-    a->annot->setAppearanceState (ustate);
-    g_free (ustate);
-  }
-#endif
-  // This function is in the library, but the enforced date parsing is
-  // incomplete (at least in some versions), because it ignores the
-  // timezone.
```

Comment:
Regarding these three lines, it seems to be referencing a bug in poppler-glib that was fixed **13 years ago**. As seen with this commit: https://gitlab.freedesktop.org/poppler/poppler/-/commit/e5f5ea01d0da4c4d877b93755523dc2a7f2ec049